src/models/User.ts

```diff
@@ -75,28 +75,50 @@ constructor(json: IUser) {
     */
     'patientId'?: string;
     /**
-    * Id of the patient if the user is a patient
-    */
+     * Id of the patient if the user is a patient
+     */
     'deviceId'?: string;
+  /**
+   * Delegations that are automatically generated client side when a new database object is created by this user
+   */
+  'autoDelegations': { [key in DelegationTag]: Set<string>; };
+  /**
+   * email address of the user (used for token exchange or password recovery).
+   */
+  'email'?: string;
     /**
-    * Delegations that are automatically generated client side when a new database object is created by this user
-    */
-    'autoDelegations': { [key: string]: Set<string>; };
-    /**
-    * email address of the user (used for token exchange or password recovery).
-    */
-    'email'?: string;
-    /**
-    * mobile phone of the user (used for token exchange or password recovery).
-    */
+     * mobile phone of the user (used for token exchange or password recovery).
+     */
     'mobilePhone'?: string;
-    /**
-    * Encrypted and time-limited Authentication tokens used for inter-applications authentication
-    */
-    'authenticationTokens': { [key: string]: AuthenticationToken; };
+  /**
+   * Encrypted and time-limited Authentication tokens used for inter-applications authentication
+   */
+  'authenticationTokens': { [key: string]: AuthenticationToken; };
 
 }
 
+export type DelegationTag =
```

Comment:
In this enum, let's only use the following ones : 
```
all,
administrativeData,
generalInformation,
financialInformation,
medicalInformation,
sensitiveInformation,
confidentialInformation
```
The other values are very specific to Belgium codes, let's try to not expose it first. We can expose it later if needed